### ISSUE TITLE
bigip pool - Path defined in monitor name not being used.

### ISSUE THREAD

**Issue by rblair1000** [F5Networks/f5-ansible#362]
##### ISSUE TYPE

 - Bug Report

##### COMPONENT NAME

bigip_pool

##### ANSIBLE VERSION

```
ansible 2.4.0.0
  config file = None
  configured module search path = [u'/root/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/site-packages/ansible
  executable location = /bin/ansible
  python version = 2.7.5 (default, May  3 2017, 07:55:04) [GCC 4.8.5 20150623 (Red Hat 4.8.5-14)]

```

##### PYTHON VERSION

```
Python 2.7.5
```

##### BIGIP VERSION

```
 Product     BIG-IP
  Version     12.1.2
  Build       1.0.271
  Edition     Hotfix HF1

```

##### OS / ENVIRONMENT

Red Hat Enterprise Linux Server release 7.4 (Maipo)

##### SUMMARY

Issue with parsing the partition name from the monitor name. Appears the partition name defined in the monitor path is being strip off and the "partition" value being used instead. Specifying a monitor from the Common partition (/Common/http) but tries to find the monitor in the Prod partition.

##### STEPS TO REPRODUCE

```
- hosts: bigip
  connection: local

  vars:
    f5_device: "x.x.x.x"
    f5_user: "****"
    f5_password: "******"
    f5_partition: "PROD"
    f5_pool_name: 'new-pool-test'
    f5_pool_monitors: ['/Common/http']
    f5_pool_monitor_type: 'and_list'

  tasks:
    - name: AddPool
      bigip_pool:
        server={{ f5_device }}
        user={{ f5_user }}
        password={{ f5_password }}
        validate_certs=no
        state=present
        monitor_type={{ f5_pool_monitor_type }}
        monitors={{ f5_pool_monitors }}
        partition={{ f5_partition }}
        name={{ f5_pool_name }}
      delegate_to: localhost

```

##### EXPECTED RESULTS

Pool to be created using a monitor from the Common partition: /Common/http

##### ACTUAL RESULTS

Playbook fails:

   "msg": "400 Unexpected Error: Bad Request for uri: https://x.x.x.x:443/mgmt/tm/ltm/pool/~PROD~new-pool-test/\nText: u'{\"code\":400,\"message\":\"01070022:3: The monitor template /PROD/http was not found.\",\"errorStack\":[],\"apiError\":3}'"

```
ansible-playbook 2.4.0.0
  config file = None
  configured module search path = [u'/root/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/site-packages/ansible
  executable location = /bin/ansible-playbook
  python version = 2.7.5 (default, May  3 2017, 07:55:04) [GCC 4.8.5 20150623 (Red Hat 4.8.5-14)]
No config file found; using defaults
Parsed /tempwork/ansible/hosts inventory source with ini plugin

PLAYBOOK: f5-add-pool.bad.yml ***************************************************************************************************
1 plays in f5-add-pool.bad.yml

PLAY [bigip] ********************************************************************************************************************

TASK [Gathering Facts] **********************************************************************************************************
Using module file /usr/lib/python2.7/site-packages/ansible/modules/system/setup.py
<x.x.x.x> ESTABLISH LOCAL CONNECTION FOR USER: root
<x.x.x.x> EXEC /bin/sh -c 'echo ~ && sleep 0'
<x.x.x.x> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /root/.ansible/tmp/ansible-tmp-1507751302.04-104367052682480 `" && echo ansible-tmp-1507751302.04-104367052682480="` echo /root/.ansible/tmp/ansible-tmp-1507751302.04-104367052682480 `" ) && sleep 0'
<x.x.x.x> PUT /tmp/tmprllzdj TO /root/.ansible/tmp/ansible-tmp-1507751302.04-104367052682480/setup.py
<x.x.x.x> EXEC /bin/sh -c 'chmod u+x /root/.ansible/tmp/ansible-tmp-1507751302.04-104367052682480/ /root/.ansible/tmp/ansible-tmp-1507751302.04-104367052682480/setup.py && sleep 0'
<x.x.x.x> EXEC /bin/sh -c '/usr/bin/python /root/.ansible/tmp/ansible-tmp-1507751302.04-104367052682480/setup.py; rm -rf "/root/.ansible/tmp/ansible-tmp-1507751302.04-104367052682480/" > /dev/null 2>&1 && sleep 0'
ok: [x.x.x.x]
META: ran handlers

TASK [AddPool] ******************************************************************************************************************
task path: /tempwork/ansible/f5-add-pool.bad.yml:14
Using module file /usr/lib/python2.7/site-packages/ansible/modules/network/f5/bigip_pool.py-bu
<localhost> ESTABLISH LOCAL CONNECTION FOR USER: root
<localhost> EXEC /bin/sh -c 'echo ~ && sleep 0'
<localhost> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /root/.ansible/tmp/ansible-tmp-1507751302.75-255444160567653 `" && echo ansible-tmp-1507751302.75-255444160567653="` echo /root/.ansible/tmp/ansible-tmp-1507751302.75-255444160567653 `" ) && sleep 0'
<localhost> PUT /tmp/tmph6BUzO TO /root/.ansible/tmp/ansible-tmp-1507751302.75-255444160567653/bigip_pool.py-bu
<localhost> EXEC /bin/sh -c 'chmod u+x /root/.ansible/tmp/ansible-tmp-1507751302.75-255444160567653/ /root/.ansible/tmp/ansible-tmp-1507751302.75-255444160567653/bigip_pool.py-bu && sleep 0'
<localhost> EXEC /bin/sh -c '/usr/bin/python /root/.ansible/tmp/ansible-tmp-1507751302.75-255444160567653/bigip_pool.py-bu; rm -rf "/root/.ansible/tmp/ansible-tmp-1507751302.75-255444160567653/" > /dev/null 2>&1 && sleep 0'
The full traceback is:
  File "/tmp/ansible_b0DTL1/ansible_module_bigip_pool.py", line 733, in main
    results = mm.exec_module()
  File "/tmp/ansible_b0DTL1/ansible_module_bigip_pool.py", line 456, in exec_module
    raise F5ModuleError(str(e))

fatal: [x.x.x.x -> localhost]: FAILED! => {
    "changed": false,
    "failed": true,
    "invocation": {
        "module_args": {
            "description": null,
            "host": null,
            "lb_method": null,
            "monitor_type": "and_list",
            "monitors": [
                "/Common/http"
            ],
            "name": "new-pool-test",
            "partition": "PROD",
            "password": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "port": null,
            "quorum": null,
            "reselect_tries": null,
            "server": "x.x.x.x",
            "server_port": 443,
            "service_down_action": null,
            "slow_ramp_time": null,
            "state": "present",
            "user": "admin",
            "validate_certs": false
        }
    },
    "msg": "400 Unexpected Error: Bad Request for uri: https://x.x.x.x:443/mgmt/tm/ltm/pool/~PROD~new-pool-test/\nText: u'{\"code\":400,\"message\":\"01070022:3: The monitor template /PROD/http was not found.\",\"errorStack\":[],\"apiError\":3}'"
}
        to retry, use: --limit @/tempwork/ansible/f5-add-pool.bad.retry

PLAY RECAP **********************************************************************************************************************
x.x.x.x              : ok=1    changed=0    unreachable=0    failed=1
```

**Comment by caphrim007** [comment]
@rblair1000 there is an updated module here that I am in the process of upstreaming.

https://github.com/F5Networks/f5-ansible/blob/devel/library/bigip_pool.py

Can you let me know if the issue persists with that copy of the module?

**Comment by rblair1000** [comment]
I copied the module over getting the same error.

Thanks



Sent from my Verizon, Samsung Galaxy smartphone


-------- Original message --------

**Comment by caphrim007** [comment]
@rblair1000 this should now be fixed

**Comment by rblair1000** [comment]
Looks like its working.

Thank You



Sent from my Verizon, Samsung Galaxy smartphone


-------- Original message --------

**Comment by caphrim007** [comment]
@rblair1000  fixes were merged upstream to ansible and will be released with v2.5
